staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -248,7 +255,14 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			}
 			metrics.WatchEvents.WithContext(req.Context()).WithLabelValues(kind.Group, kind.Version, kind.Kind).Inc()
 
-			if err := s.EmbeddedEncoder.Encode(event.Object, buf); err != nil {
+			if event.Type == watch.Bookmark && utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
```

Comment:
Checking quickly if this will somehow mess up with https://github.com/kubernetes/kubernetes/pull/120300 but I think it doesn't make it visibly harder.

Might be worth checking additionally.

Reply:
yeah, it doesn't it will be an easy merge :)